tests/tests/gas_price.rs

```diff
@@ -410,9 +523,419 @@ async fn startup__can_override_gas_price_values_by_changing_config() {
+        "--debug",
+        "--poa-instant",
+        "true",
+        "--min-da-gas-price",
+        "100",
+    ];
+    let rt = tokio::runtime::Runtime::new().unwrap();
+    let temp_dir = rt.block_on(async {
+        let driver = FuelCoreDriver::spawn(&args).await.unwrap();
+        driver.client.produce_blocks(100, None).await.unwrap();
+        tokio::time::sleep(Duration::from_millis(100)).await;
+        driver.kill().await
+    });
+
+    // rollback 50 blocks
+    let temp_dir = rt.block_on(async {
+        let driver = FuelCoreDriver::spawn_with_directory(temp_dir, &args)
+            .await
+            .unwrap();
+        for _ in 0..50 {
+            driver
+                .node
+                .shared
+                .database
+                .gas_price()
+                .rollback_last_block()
+                .unwrap();
+            let gas_price_db_height = driver
+                .node
+                .shared
+                .database
+                .gas_price()
+                .latest_height()
+                .unwrap();
+            tracing::info!("gas price db height: {:?}", gas_price_db_height);
+        }
+        driver.kill().await
+    });
+
+    // when
+    // restart node
+    rt.block_on(async {
+        let driver = FuelCoreDriver::spawn_with_directory(temp_dir, &args)
+            .await
+            .unwrap();
+        let onchain_db_height = driver
+            .node
+            .shared
+            .database
+            .on_chain()
+            .latest_height_from_metadata()
+            .unwrap()
+            .unwrap();
+        let gas_price_db_height = driver
+            .node
+            .shared
+            .database
+            .gas_price()
+            .latest_height()
+            .unwrap();
+        assert_eq!(onchain_db_height, gas_price_db_height);
+    });
+}
+
+fn node_config_with_da_committer_url(url: &str) -> Config {
+    let block_gas_limit = 3_000_000;
+    let chain_config = ChainConfig {
+        consensus_parameters: ConsensusParameters::V1(ConsensusParametersV1 {
+            block_gas_limit,
+            ..Default::default()
+        }),
+        ..ChainConfig::local_testnet()
+    };
+    let mut node_config =
+        Config::local_node_with_configs(chain_config, StateConfig::local_testnet());
+    let starting_gas_price = 10_000_000;
+    node_config.block_producer.coinbase_recipient = Some([5; 32].into());
+    node_config.min_da_gas_price = starting_gas_price;
+    node_config.max_da_gas_price = u64::MAX;
+    node_config.max_da_gas_price_change_percent = 15;
+    node_config.block_production = Trigger::Never;
+    node_config.da_committer_url = Some(url.to_string());
+    node_config.da_poll_interval = Some(100);
+    node_config.da_gas_price_p_component = 224_000;
+    node_config.da_gas_price_d_component = 2_690_000;
+    node_config.block_activity_threshold = 0;
+    node_config
+}
+
+#[test]
+fn produce_block__algorithm_recovers_from_divergent_profit() {
```

Comment:
This is a very expensive test. Do we want to always run it?

Should we add it to `nexttest` configuration as a "slow" test?

Reply:
will address in a follow up pr *maybe*